framework/src/systems/NonlinearSystemBase.C

```diff
@@ -1368,9 +1369,13 @@ NonlinearSystemBase::computeResidualInternal(const std::set<TagID> & tags)
     ConstElemRange & elem_range = *_mesh.getActiveLocalElementRange();
 
     ComputeResidualThread cr(_fe_problem, tags);
-
     Threads::parallel_reduce(elem_range, cr);
 
+    using FVRange = StoredRange<std::vector<FaceInfo>::const_iterator, FaceInfo>;
```

Comment:
Should we check if we have any FV kernels? If no kernels, we do not need to do calculation

Reply:
We could, but the check happens inside the flux loop itself which, I think, is a better place for it.  This isn't a particularly performance sensitive location for avoiding the creation of a loop object since we are (obviously) outside the loop already.  I'd just assume keep the check deeper and closer to the code that understands it so we are less likely to have it go out of sync with the actual criteria it should be checking.